Sources/MuxUploadSDK/Upload/ChunkedFileUploader.swift

```diff
@@ -88,15 +89,28 @@ class ChunkedFileUploader {
             do {
                 // It's fine if it's already open, that's handled by ignoring the call
                 try file.openFile(fileURL: uploadInfo.videoFile)
+                let fileSize = file.fileSize
                 try file.seekTo(byte: lastByte)
                 let result = try await makeWorker().performUpload()
                 file.close()
-                
+
                 let success = UploadResult(
                     finalProgress: result.progress,
                     startTime: result.startTime,
                     finishTime: result.updateTime
                 )
+
+                let asset = AVAsset(url: uploadInfo.videoFile)
+
+                var duration: CMTime
+                if #available(iOS 16, *) {
+                    duration = try await asset.load(.duration)
+                } else {
+                    duration = asset.duration
+                }
```

Comment:
I worry that this could cause the SDK to hang while it calculates the duration. Is there a better way to do this or is it ok?

Reply:
What do you think of using [this](https://developer.apple.com/documentation/avfoundation/avasynchronouskeyvalueloading/1387321-loadvaluesasynchronously) when running on an iOS version predating 15? ([Have a separate comment about the version check.](https://github.com/muxinc/swift-upload-sdk/pull/3/files#r1147035797))

Reply:
You're right! Xcode told me it's deprecated in iOS 16, but now I see it was first available in iOS 15.